### PR TITLE
Removes manual domain name checking

### PR DESCRIPTION
**Pull request by kbamponsem**
This PR contains updates to the telemetry `ureq_dns_resolver` module which resolves domain names to a Fixed IP. 

It removes the manual certificate validation module.

### CODE REVIEW

File: src/ureq_dns_resolver.rs [hunk 1]
```diff
@@ -140,17 +36,11 @@ impl InternalAgent {
...
 
-        // Create the CertificateVerifierNoHostname object
-        let cert_verifier = CertificateVerifierNoHostname::try_from((
-            &webpki_roots::TLS_SERVER_ROOTS,
-            allowed_dns_servers.iter().map(|s| s.to_string()).collect(),
-        ))
-        .unwrap();
-
         // See rustls documentation for more configuration options.
         let tls_config = rustls::ClientConfig::builder()
```
Comment (JoFrost) [inline src/ureq_dns_resolver.rs:40]: Do you really need to specify a custom TLS config? Can't you use the default one?

File: src/telemetry.rs [hunk 2]
```diff
@@ -216,7 +216,7 @@ impl Telemetry {
             setup()?;
 
             // Create a new agent from the InternalAgent struct
-            let agent = InternalAgent::new(TELEMETRY_IP, "443", &["telemetry.mithrilsecurity.io"]);
+            let agent = InternalAgent::new(TELEMETRY_IP, "443");
```
Comment (JoFrost) [inline src/telemetry.rs:219]: Where do you use the telemetry URL?

File: src/ureq_dns_resolver.rs [hunk 3]
```diff
@@ -1,19 +1,13 @@
...
+
     //! A DNS resolver that always returns the same value regardless of
     //! the host it was queried for.
-
+    //!
+    //! NB: The following modules `fixed_resolver` is from this website:
+    //! https://scvalex.net/posts/48/
     use std::net::SocketAddr;
 
     pub struct FixedResolver(pub SocketAddr);
```
Comment (JoFrost) [inline src/ureq_dns_resolver.rs:13]: I don't see the usage of the URL anywhere. Where is it used?

File: src/telemetry.rs (hunk 2 above)
Comment (kbamponsem) [reply src/telemetry.rs:219]: The URL is used in here :https://github.com/mithril-security/blindai/blob/telemetry/src/telemetry.rs#L163.

File: src/ureq_dns_resolver.rs (hunk 3 above)
Comment (kbamponsem) [reply src/ureq_dns_resolver.rs:13]: The URL is used to make the post req: https://github.com/mithril-security/blindai/blob/telemetry/src/telemetry.rs#L163

File: src/ureq_dns_resolver.rs (hunk 1 above)
Comment (kbamponsem) [reply src/ureq_dns_resolver.rs:40]: Probably, but I am looking into the rustls docs to see.

**Pull request merged**